packages/adblocker/src/engine/engine.ts

```diff
@@ -1499,13 +1527,115 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
         if (result.filter !== undefined) {
           result.exception = this.exceptions.match(request, this.isFilterExcluded.bind(this));
         }
+
+        // If the request is allowed in any way, we match request removeparam.
+        if (result.filter === undefined) {
+          const searchParamSeparatorIndex = request.url.indexOf('?');
+          if (
+            searchParamSeparatorIndex !== -1 ||
+            searchParamSeparatorIndex !== request.url.length - 1
```

Comment:
Should this be (otherwise I think it's always evaluated to `true`):
```suggestion
            searchParamSeparatorIndex !== -1 &&
            searchParamSeparatorIndex !== request.url.length - 1
```

If that change is correct, let's add a unit test to cover that.

Reply:
fixed in 8f87db81a3564ae1c59c4ea934a99c3e25cbec0a